onnxruntime/contrib_ops/cpu/grid_sample.cc

```diff
@@ -1,273 +1,21 @@
-                }
-              }  // out of bound
-
-              if (mode_ == Nearest) {
-                // x, y are integers in all padding modes
-                *Y_gridpoint = PixelAtGrid(X_data, static_cast<int64_t>(y), static_cast<int64_t>(x), H_in, W_in, border);
-                continue;
-              }
-
-              if (mode_ == Bilinear) {
-                int64_t x1 = static_cast<int64_t>(std::floor(x));
-                int64_t y1 = static_cast<int64_t>(std::floor(y));
-                int64_t x2 = x1 + 1;
-                int64_t y2 = y1 + 1;
-
-                T p11 = PixelAtGrid(X_data, y1, x1, H_in, W_in, border);
-                T p12 = PixelAtGrid(X_data, y1, x2, H_in, W_in, border);
-                T p21 = PixelAtGrid(X_data, y2, x1, H_in, W_in, border);
-                T p22 = PixelAtGrid(X_data, y2, x2, H_in, W_in, border);
-
-                T dx2 = static_cast<T>(x2) - x;
-                T dx1 = x - static_cast<T>(x1);
-                T dy2 = static_cast<T>(y2) - y;
-                T dy1 = y - static_cast<T>(y1);
-                *Y_gridpoint = dy2 * (dx2 * p11 + dx1 * p12) + dy1 * (dx2 * p21 + dx1 * p22);
-              }
-              if (mode_ == Bicubic) {
-                int64_t x0 = static_cast<int64_t>(std::floor(x)) - 1;  // top-left corner of the bbox
-                int64_t y0 = static_cast<int64_t>(std::floor(y)) - 1;
-                T p[4][4] = {};  // [H][W]
-                for (int64_t h = 0; h < 4; h++) {
-                  for (int64_t w = 0; w < 4; w++) {
-                    p[h][w] = PixelAtGrid(X_data, h + y0, w + x0, H_in, W_in, border);
-                  }
-                }
-                T dx = static_cast<T>(x - x0 - 1);
-                T dy = static_cast<T>(y - y0 - 1);
-                *Y_gridpoint = GsBicubicInterpolate(p, static_cast<float>(dx), static_cast<float>(dy));
-              }
-            }
-          }
-        });
-  }
-  return Status::OK();
-}
-
-#define REGISTER_KERNEL_TYPED(T)                                   \
-  ONNX_OPERATOR_TYPED_KERNEL_EX(                                   \
-      GridSample,                                                  \
-      kMSDomain,                                                   \
-      1,                                                           \
-      T,                                                           \
-      kCpuExecutionProvider,                                       \
-      KernelDefBuilder()                                           \
-          .TypeConstraint("T1", DataTypeImpl::GetTensorType<T>())  \
-          .TypeConstraint("T2", DataTypeImpl::GetTensorType<T>()), \
+#define REGISTER_KERNEL_TYPED(T)                                    \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                    \
+      GridSample,                                                   \
+      kMSDomain,                                                    \
```

Comment:
Is this supposed to be ONNX domain now?

Reply:
Ok, I see the ONNX domain registration is in a different file later on. Is this to maintain backward compatibility? Thanks!